### ISSUE TITLE
@MonotonicNonNull not respected for lazy initialization

### ISSUE THREAD

**Issue by mizosoft** [uber/NullAway#382]
From my understanding, checkerframework's `@MonotonicNonNull` is the equivalent of errorprone's `@LazyInit`. However, nullaway respects the latter but ignores the former. For example running nullaway on this code: 
```java
public class MemoizingSupplier implements Supplier<String> {
  private final Supplier<String> supplier;
  private @MonotonicNonNull String val;

  public MemoizingSupplier(Supplier<String> supplier) {
    this.supplier = supplier;
  }

  @Override public String get() {
    if (val == null) {
      val = supplier.get();
    }
    return val;
  }
}
```
The compiler issues the following warning: 
```
warning: [NullAway] initializer method does not guarantee 
@NonNull field val is initialized along all control-flow paths (remember to check for exceptions or early returns).
    public MemoizingSupplier(Supplier<String> supplier) {
           ^
```
As previously noted, changing to `@LazyInit` fixes the warning. Is it that nullaway doesn't integrate well with cf's annotations ?

**Comment by lazaroclapp** [comment]
I think you are correct. It might be worthwhile to add that annotation to the default "excluded from initialization checking" annotations, and we'll look into it.

In the meantime, the set of such annotations is a [configuration option](https://github.com/uber/NullAway/wiki/Configuration#excluded-field-annotations) for NullAway. You can try passing: 

```
-XepOpt:NullAway:ExcludedFieldAnnotations=
      org.checkerframework.checker.nullness.qual.MonotonicNonNull
``` 

to have it treated exactly as `@LazyInit`.

In fact, that's how we handle `@LazyInit`, it's just a default prepended value for that same option.

If that works for you, let me know, and we can maybe make that one part of the default known annotations for the next NullAway release too.

**Comment by mizosoft** [comment]
Hi @lazaroclapp, thanks for the reply. Adding `@MonotonicNonNull` to that option indeed worked for me. Thanks for letting me know `@LazyInit` is a default for that option. I thought it was handled in some special way!

I'm guessing in that case that also recognizing `@MonotonicNonNull` would be trivial. I think doing so is a good idea as cf's annotations is now the preferred option after people have migrated from jsr305 so it should be recognized by nullaway.

**Comment by lazaroclapp** [comment]
So, making it the default sounds good to me. Particularly after reading the only two rules for `@MonotonicNonNull` [here](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/MonotonicNonNull.html), which seem to match NullAway's behavior for "ExcludedFieldAnnotations".

I think this doesn't merit cutting a new release (let me know if that isn't true), but it should be the default in the next NullAway release (assuming #383 lands, which is likely).
